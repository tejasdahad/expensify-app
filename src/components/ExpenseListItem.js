import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    <div>
        <p>Description: {description}</p>
        <p>Amount: {amount}</p>
        <p>Created At: {createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense({id}))
        }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);
