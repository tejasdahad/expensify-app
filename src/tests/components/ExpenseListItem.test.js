import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with a expense', () => {
    const expense = {
        id: '1',
        description: 'rent',
        note: '',
        amount: 100,
        createdAt: 1000
    };
    const wrapper = shallow(<ExpenseListItem {...expense} />);
    expect(wrapper).toMatchSnapshot();
});