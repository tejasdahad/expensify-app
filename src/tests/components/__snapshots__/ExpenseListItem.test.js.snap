// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with a expense 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      rent
    </h3>
  </Link>
  <p>
    $1.00
    -
    January 1st, 1970
  </p>
</div>
`;
